bpo-39573: Add Py_IS_TYPE macro (#18488)
python/cpython: Misc/NEWS.d/next/Core and Builtins/2020-02-13-01-30-22.bpo-39573.uTFj1m.rst

> Add :c:func:`Py_IS_TYPE` static inline function to check whether

Reviewer: Can you please check if the following formatting is rendered properly in HTML? (``make -C Doc venv html``).

```suggestion
Add :c:func:`Py_IS_TYPE(o, type)` static inline function to check whether
```

> Add :c:macro:`Py_IS_TYPE` macro to check whether the type of object *o* is

Reviewer: ```suggestion
Add :c:func:`Py_IS_TYPE(o, type)` static inline function to check whether the type of object *o* is
```

> the type of object *o* is the *type*.

Reviewer: maybe also here:

```suggestion
the object *o* type is *type*.
```
